x-pack/plugins/ml/public/data_frame/pages/data_frame_new_pivot/directive.tsx

```diff
@@ -16,22 +16,31 @@ import { I18nContext } from 'ui/i18n';
 // @ts-ignore
 import { SearchItemsProvider } from '../../../jobs/new_job/utils/new_job_utils';
 
-import { IndexPatternContext } from '../../common';
+import { KibanaContext } from '../../common';
 import { Page } from './page';
 
-module.directive('mlNewDataFrame', ($route: any, Private: any) => {
+module.directive('mlNewDataFrame', ($injector: any, $route: any, Private: any) => {
```

Comment:
nit, `indexPatterns` and `config` could be added to the list of injected parameters. rather than having a mix

Reply:
👍 Refactored with a bit more extensive TS in 122092c.